pkg/validation/commands.go

```diff
@@ -11,21 +11,21 @@ import (
 // 1. there are no duplicate command ids
 // 2. the command type is not invalid
 // 3. if a command is part of a command group, there is a single default command
-func ValidateCommands(commands []v1alpha2.Command, components []v1alpha2.Component) (err error) {
+func ValidateCommands(commands []v1alpha2.Command, components []v1alpha2.Component) (errList []error) {
```

Comment:
Note that there is also the [`multierror` package](https://github.com/hashicorp/go-multierror), which is already in our `go.mod` -- this might make the errors a little easier to use in calling code.

Reply:
We want to return `[]error` instead of `error` (https://github.com/devfile/api/issues/577#issuecomment-896172384)
if concatenating all errors into a single error, it will be hard to split those errors and identify each error type

Reply:
Makes sense to me, thanks!